### ISSUE TITLE
GPU COUNT and BATCH SIZE not work

### ISSUE THREAD

**Issue by kuonangzhe** [multimodallearning/pytorch-mask-rcnn#7]
Thanks for the great work!
The problem I met is on gpu and batch size. 

The GPU_COUNT only affects the number of images processed in each epoch, but there is only one gpu that is really working. The MaskRCNN module was not implemented with DataParallel, while it CANNOT be paralleled on various gpus as I tried. 

The BATCH_SIZE only affects the number of images in each epoch, but in dataloader, the setting for each batch is again 1. When setting to 2 or higher, there will be error for not dealling with data shapes to make them batchable.

Is there any plan to fix these issues? Thank you so much!

**Comment by hao522** [comment]
Same question here

**Comment by lasseha** [comment]
As i only own a GTX 1070 that can fit no more than one image, only a batch size of 1 is supported right now. Therefore, GPU_COUNT may only be 0 or 1 and BATCH_SIZE is a workaround to resemble the matterport implementation by sequential batch sizing. Contributions supporting multiple gpus and arbitrary batch sizes are very welcome.

**Comment by skrish13** [comment]
just a small doubt @lasseha, have you provided code for training from scratch? wasnt able to find through a brief glance... (apologies for not going through the repo in detail)

**Comment by lasseha** [comment]
The training routine is implemented but i could not achieve the matterport results yet. Results are about 8 percent off. There must be either a mistake in the training code or i did not find the right training schedule yet.

**Comment by skrish13** [comment]
ah its inside [model.py](https://github.com/multimodallearning/pytorch-mask-rcnn/blob/master/model.py). Nice, thanks!

**Comment by kuonangzhe** [comment]
@lasseha Sure, thanks for the help!
